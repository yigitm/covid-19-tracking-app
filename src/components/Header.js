import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Alert } from 'react-bootstrap';

const Header = () => {
  const totalCase = useSelector((state) => state.headerReducers);

  return totalCase.map((covid) => (
    <Alert variant="success">
      <Alert.Heading className="p-5 text-center">
        Total Number of Cases:{covid.total}
      </Alert.Heading>
    </Alert>
  ));
};
export default Header;
