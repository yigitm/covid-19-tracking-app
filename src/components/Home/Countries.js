import React from 'react';
import { useSelector } from 'react-redux';
import { Card } from 'react-bootstrap';
import { Link } from 'react-router-dom';

const Countries = () => {
  const countries = useSelector((state) => state.countryReducers);

  return countries.map((country, index) => (
    <Card key={country.id} className="w-50 gap-0 p-0 border-0">
      <Card.Body className={(index + 1) % 4 >= 2 ? 'bg-red' : 'bg-red-bold'}>
        <div className="d-flex justify-content-between">
          <i className="fas fa-chart-pie text-left text-light icon" />
          <Link to={country.id}>
            <i className="far fa-arrow-alt-circle-right icon text-light" />
          </Link>
        </div>
        <Card.Title className="text-end text-light">
          <p className="font-big">{country.name}</p>
        </Card.Title>
        <Card.Text className="text-end text-light font-small">
          Total Cases: {country.total}
        </Card.Text>
        <Card.Text className="text-end text-light font-small">
          New Cases: {country.newCase}
        </Card.Text>
      </Card.Body>
    </Card>
  ));
};
export default Countries;
