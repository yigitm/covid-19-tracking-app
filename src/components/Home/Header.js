import React from 'react';
import { useSelector } from 'react-redux';
import { Col, Row } from 'react-bootstrap';

const Header = () => {
  const totalCase = useSelector((state) => state.headerReducers.total);

  return (
    <Row className="bg-red">
      <Col className="col-5 p-4 mx-auto">
        <i className="fas fa-globe-americas icon-header text-light m-5" />
      </Col>
      <Col className="col-5 p-4 ">
        <p className="text-light mt-5">
          <span className="font-header text-left">GLOBAL</span>
          <br />
          <span className="font-header">
            {' '}
            {totalCase}
            {' '}
            cases
          </span>
        </p>
      </Col>
      <Col className="col-12 p-2 text-left font-big text-light text-break bg-red-bold">
        Covid-19 Stats
      </Col>
    </Row>
  );
};
export default Header;
