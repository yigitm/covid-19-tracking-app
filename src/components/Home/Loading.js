import React from 'react';
import Spinner from 'react-bootstrap/Spinner';

const Loading = () => (
  <div className="d-flex justify-content-center">
    <Spinner animation="grow" />
    <Spinner animation="grow" />
    <Spinner animation="grow" />
    <h1>Loading...</h1>
  </div>
);
export default Loading;
