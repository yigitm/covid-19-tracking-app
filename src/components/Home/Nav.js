import React from 'react';
import { Navbar } from 'react-bootstrap';
import { Link } from 'react-router-dom';

const Nav = () => (
  <Navbar className="p-0">
    <Navbar.Brand>
      <Link to="/">
        <i className="fas fa-angle-left"></i>
      </Link>
    </Navbar.Brand>
    <Navbar.Toggle />
    <Navbar.Collapse className="justify-content-end">
      <Navbar.Text>
        <i className="fas fa-microphone"></i> <i className="fas fa-cog"></i>
      </Navbar.Text>
    </Navbar.Collapse>
  </Navbar>
);

export default Nav;
