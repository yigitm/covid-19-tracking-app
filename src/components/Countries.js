import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Card, Col } from 'react-bootstrap';
import Loading from './Loading';

const Countries = () => {
  const countries = useSelector((state) => state.countryReducers);

  return countries.map((country) => (
    <Col key={country.id} className="w-100" className="col">
      <Card.Body>
        <div className="d-flex justify-content-between">
          <i class="fas fa-chart-pie fa-3x text-left" />
          <i class="far fa-arrow-alt-circle-right fa-2x" />
        </div>
        <Card.Title className="text-end">{country.name}</Card.Title>
        <Card.Text className="text-end">Total Cases: {country.total}</Card.Text>
        <Card.Text className="text-end">New Cases: {country.newCase}</Card.Text>
      </Card.Body>
    </Col>
  ));
};
export default Countries;
