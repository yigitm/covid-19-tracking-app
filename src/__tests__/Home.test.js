import React from 'react';
import { Container, Row } from 'react-bootstrap';
import Loading from '../components/Home/Loading';
import Nav from '../components/Home/Nav';
import Header from '../components/Home/Header';
import Countries from '../components/Home/Countries';

describe('Test Home render', () => {
  test('Test Home', () => {
    const Home = () => (
      <Container fluid className="bg-red-bold">
        <Loading />
        <Nav />
        <Header />
        <Row>
          <Countries />
        </Row>
      </Container>
    );

    expect(Home()).toMatchSnapshot();
  });
});
