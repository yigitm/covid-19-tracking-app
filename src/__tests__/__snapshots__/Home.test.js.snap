// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home render Test Home 1`] = `
<Container
  className="bg-red-bold"
  fluid={true}
>
  <Loading />
  <Nav />
  <Header />
  <Row>
    <Countries />
  </Row>
</Container>
`;
